src/test/java/org/opensearch/knn/index/mapper/KNNVectorFieldMapperTests.java

```diff
@@ -526,6 +549,90 @@ public void testKNNVectorFieldMapper_merge_fromModel() throws IOException {
     }
 
+    public void testLuceneFieldMapper_parseCreateField_docValues() throws IOException {
+        // Create a lucene field mapper that creates a binary doc values field as well as KnnVectorField
+        KNNMethodContext knnMethodContext = new KNNMethodContext(
+            KNNEngine.LUCENE,
+            SpaceType.DEFAULT,
+            new MethodComponentContext(METHOD_HNSW, Collections.emptyMap())
+        );
+
+        KNNVectorFieldMapper.KNNVectorFieldType knnVectorFieldType = new KNNVectorFieldMapper.KNNVectorFieldType(
+            TEST_FIELD_NAME,
+            Collections.emptyMap(),
+            TEST_DIMENSION,
+            knnMethodContext
+        );
+
+        LuceneFieldMapper.CreateLuceneFieldMapperInput.CreateLuceneFieldMapperInputBuilder inputBuilder =
+            LuceneFieldMapper.CreateLuceneFieldMapperInput.builder()
+                .name(TEST_FIELD_NAME)
+                .mappedFieldType(knnVectorFieldType)
+                .multiFields(FieldMapper.MultiFields.empty())
+                .copyTo(FieldMapper.CopyTo.empty())
+                .hasDocValues(true)
+                .ignoreMalformed(new Explicit<>(true, true))
+                .knnMethodContext(knnMethodContext);
+
+        LuceneFieldMapper luceneFieldMapper = createMockLuceneFieldMapper(inputBuilder.build(), TEST_VECTOR);
+
+        ParseContext.Document document = new ParseContext.Document();
+        ContentPath contentPath = new ContentPath();
+        ParseContext parseContext = mock(ParseContext.class);
+        when(parseContext.doc()).thenReturn(document);
+        when(parseContext.path()).thenReturn(contentPath);
+
+        luceneFieldMapper.parseCreateField(parseContext, TEST_DIMENSION);
+
+        // Document should have 2 fields: one for VectorField (binary doc values) and one for KnnVectorField
+        List<IndexableField> fields = document.getFields();
+        assertEquals(2, fields.size());
+        IndexableField field1 = fields.get(0);
+        IndexableField field2 = fields.get(1);
+
+        VectorField vectorField;
+        KnnVectorField knnVectorField;
+        if (field1 instanceof VectorField) {
+            assertTrue(field2 instanceof KnnVectorField);
+            vectorField = (VectorField) field1;
+            knnVectorField = (KnnVectorField) field2;
+        } else {
+            assertTrue(field1 instanceof KnnVectorField);
+            assertTrue(field2 instanceof VectorField);
+            knnVectorField = (KnnVectorField) field1;
+            vectorField = (VectorField) field2;
+        }
+
+        assertEquals(TEST_VECTOR_BYTES_REF, vectorField.binaryValue());
+        assertArrayEquals(TEST_VECTOR, knnVectorField.vectorValue(), 0.001f);
+
+        // Test when doc values are disabled
+        inputBuilder.hasDocValues(false);
+        luceneFieldMapper = createMockLuceneFieldMapper(inputBuilder.build(), TEST_VECTOR);
```

Comment:
why we cannot create mock implementation with mockito?

Reply:
Maybe this is the wrong name for this function. I didnt create a mock because we are testing LuceneFieldMapper. However, I override some of the methods (i.e. checking if plugin is disabled and parsing the the vector) to make testing easier.

Reply:
It may be a good use case for [mockito spy](https://www.baeldung.com/mockito-spy), you essentially pass all the calls to the original class, but mocking one-two methods with a custom implementation. What do you think?